### DISCUSSION TITLE
How to "update" the context in a conditional tuple in the same transactional write operation?

### DISCUSSION THREAD

**Discussion by velmohan** [openfga/community#271] [Q&A] [answered]
I am facing an issue where I am not able to update the context in a condition in the same transaction.

Say for example, I have the following conditional tuple (taken from this [example](https://github.com/openfga/sample-stores/blob/main/stores/groups-resource-attributes/store.fga.yaml))

```
user: "group:marketing#member"
relation: "can_access_docs"
object: "organization:acme"
condition: 
  name: doc_viewer_condition
  context: 
    allowed_statuses : ["published"]
```

Now, I want to update the allowed_statuses to `["published", "draft"]` as follows.

```
user: "group:marketing#member"
relation: "can_access_docs"
object: "organization:acme"
condition: 
  name: doc_viewer_condition
  context: 
    allowed_statuses : ["published", "draft"]
```

Ideally I want to create a write  request which deletes the first one and writes the second one in the same transaction. But, the write request is failing with the error "duplicate tuple in write". 

How can we go about this?  I can think of 2 solutions.
1. Delete the tuple in a separate request before adding again but we lose the transactional safety.
2. Provide allowed_statuses in the check request so we will not have to maintain the context in tuples.

Are there any better ideas?

**Comment by aaguiarz** [comment, marked as answer]
Hi @velmohan 

We should process first the deletions and later the writes, so you don't get the error. For now, I'd go with Option 1..

**Reply by aaguiarz** [reply]
I created an Idea for this https://github.com/orgs/openfga/discussions/282

**Reply by velmohan** [reply]
Thank you @aaguiarz
